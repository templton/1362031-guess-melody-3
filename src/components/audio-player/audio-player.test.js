import React from 'react';
import renderer from 'react-test-renderer';

import AudioPlayer from './audio-player.jsx';

const mock = {
  song: {
    src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`
  }
};

describe(`Render audio player`, () => {
  it(`AudioPlayer is rendered correctly`, () => {
    const {song} = mock;

    const tree = renderer.create(<AudioPlayer
      src={song.src}
      isPlaying={false}
      onPlayButtonClick={() => {}}
      isLoading={false}
      progress={0}
      setPlayingStatus={() => {}}
      setLoadingStatus={() => {}}
      setProgress={() => {}}
    />, {
      createNodeMock: () => {
        return {};
      }
    }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
