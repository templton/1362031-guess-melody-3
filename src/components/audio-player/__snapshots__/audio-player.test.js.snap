// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render audio player AudioPlayer is rendered correctly 1`] = `
Array [
  <button
    className="track__button track__button--play"
    disabled={false}
    onClick={[Function]}
    type="button"
  />,
  <div
    className="track__status"
  >
    <audio
      onCanPlayThrough={[Function]}
      onPause={[Function]}
      onPlay={[Function]}
      onTimeUpdate={[Function]}
      src="https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg"
    />
  </div>,
]
`;
